buildSrc/src/main/groovy/org/gradle/testing/DistributionTest.groovy

```diff
@@ -52,118 +62,90 @@ class DistributionTest extends Test {
 
     }
 
-    /**
-     * The system properties not coming from absolute paths.
-     *
-     * We cannot rely on {@link #getSystemProperties()} as an input since it will contain absolute paths -
-     * and this defeats relocatability of the distribution test tasks.
-     * The input coming from {@link #getSystemProperties()} will be set to an empty map in the build script
-     * since we do not have the possibility yet to override it from here.
-     */
-    @Input
-    Map<String, Object> getPlainSystemProperties() {
-        (super.getSystemProperties() - fileSystemProperties.collectEntries { key, value -> [(key): value.absolutePath] }).findAll { key, value -> !ignoredSystemProperties.contains(key)}
-    }
-
-    /**
-     * SystemProperties are ignored as inputs since they contain absolute paths.
-     * See {@link #getPlainSystemProperties()} and {@link #fileSystemProperty(java.lang.String, java.io.File)} how we deal with those.
-     *
-     * {@inheritDoc}
-     */
-    @Internal
-    @Override
-    Map<String, Object> getSystemProperties() {
-        return super.getSystemProperties()
-    }
-
-    @Nested
-    GradleDistribution getDistribution() {
-        return gradleHomeDir ? new GradleDistribution(project, gradleHomeDir) : null
-    }
-
     @Internal
-    File gradleHomeDir
-
-    void setGradleHomeDir(File gradleHomeDir) {
-        this.gradleHomeDir = fileSystemProperty('integTest.gradleHomeDir', gradleHomeDir)
-    }
+    final DirectoryProperty gradleHomeDir
```

Comment:
I could, but the properties would stay `@Internal`, since we actually track something different for every single one of them.
Let me try to make this better.